### ISSUE TITLE
PrimarySmtpAddress / EmailAddresses missing ?

### ISSUE THREAD

**Issue by gitbpotalan** [microsoft/Microsoft365DSC#4572]
Hello,
Am I missing something or these information are not extracted ?
I parsed AAD and EXO but I cannot find any component with these information. 
Thanks in advance for the information !

**Comment by Borgquite** [comment]
These attributes are present on EXODistributionGroup and EXOSharedMailbox, and EXOMailContact has the ExternalEmailAddress attribute which is equivalent to PrimarySMTPAddress.

Please can we close this?

**Comment by gitbpotalan** [comment]
There is no "Get-Mailbox" equivalent in M365 DSC, the closest possible to retrieve mailboxes is EXOCASMailboxSettings.
But this component does not include PrimarySmtpAddress.
How are you supposed to accurately export a snapshot of your configuration if the important informations are missing ?

**Comment by Borgquite** [comment]
@gitbpotalan Ah, that's a fair point! Rather than using Exchange, probably the easiest way would be to update AADUser, which has a 'mail' attribute, currently unexposed, which should reflect PrimarySMTPAddress in Exchange?

**Comment by gitbpotalan** [comment]
@Borgquite Yeah sure but mail is supposed to be part of Exchange configuration on a user mailbox I guess.
By the way AADUsers cannot be restored if users are synchronized from AD onpremise environment.

**Comment by Borgquite** [comment]
@gitbpotalan Ideally yes - a new EXOMailbox, EXOMailUser (and probably EXOUser) resource should be created. I'm suggesting a quick fix that gets you what you want, for now.

For on-premises environments, the whole of Microsoft365DSC isn't going to work for restorations for attributes that are mastered on-premises. You need to use something else, like ActiveDirectoryDsc.

**Comment by gitbpotalan** [comment]
yes, i was just replying to your comment about using AADUser instead. but it won't work because of this limitation, so this is not a possible workaround.

**Comment by Borgquite** [comment]
If I add it to AADUser it will give exactly the same functionality for this attribute as creating a new EXOMailbox resource. You won't be able to restore PrimarySmtpAddress on an Exchange Online mailbox synced from on-premises, any more than you'll be able to restore Mail on an Entra user synced from on-premises.

If you want to restore attributes for on-premises objects, you'll need a different project.

**Comment by gitbpotalan** [comment]
Here's the usecase : 
-Backup each service and its configuration 
-Each configuration is a different component
-A customer wants to restore its mailboxes' configuration. PrimarySmtpAddress should be restored on several users, should I tell him "its not possible, email exists only in AADUser" - so if you want to restore email parameters, you'll have to restore AADUser + EXOMailboxSettings ?

**Comment by Borgquite** [comment]
@gitbpotalan Yes. In Microsoft 365, configuration settings for a single item are separated across many services. For your example of a mailbox (regardless of type):

- Entra has a 'user account' which includes generic attributes like the username, password, User Principal Name, and various other bits of metadata like DisplayName, Department, Office etc.
- Exchange has a 'mailbox' which includes Exchange-specific attributes like mailbox permissions etc

There are some attributes which can be set using either service (e.g. mail/PrimarySmtpAddress) and others which can only be set on one or the other. This is further complicated for hybrid accounts, where most of the Entra attributes are mastered on-premises. The Microsoft 365 admin center somewhat protects you against this complexity by giving you one set of controls, but I suspect, under the covers, it 'switches' between Graph and Exchange Online depending on what you want. If you use the workload-specific Admin centers (e.g. Entra, Exchange Online) you'll get a feel for which settings are 'owned' by which workload.

In terms of your requirement, it's quite easy to add 'mail' to the AADUser resource module (which interfaces with Entra). There are some attributes which can't / shouldn't be changed in Entra (e.g. showInAddressList/HiddenFromAddressListsEnabled, proxyAddresses, creating shared mailboxes) and for those I suggest raising another 'issue' requesting a full EXOMailbox resource. I am hoping to spend a little time fixing up the AADUser resource with Mail at some point as a quick fix. Creating an EXOMailbox resource is a larger job that I don't have time to do (to be clear, I'm a volunteer on this project, but there are actual Microsoft employees around too who may be able to get round to it if you present the use case).

So the short term fix for your immediate question is to add 'mail' to AADUser. That would allow someone to restore mailbox parameters with AADUser + EXOMailboxSettings, as you said. The longer-term desire you have (restore **all** mailbox configurations) needs a new resource, probably EXOMailbox, and I suggest you raise a new issue for that. Does that help?

**Comment by Borgquite** [comment]
@gitbpotalan I hope the next release will meet the basic need by exporting the mail attribute for AADUser. If you would like to export all mail related attributes associated with a mailbox, please do open a new issue for an EXOMailbox resource.
